draft-ietf-jsonpath-base.md

```diff
@@ -577,207 +590,227 @@ of node.
 
 ## Selectors
 
-### Dot Child Selector
+A JSONPath query consists of a sequence of selectors. Valid selectors are
+
+  * Root selector `'$'`
+  * Dot selector `'.<name>'`, used with object member names exclusively.
+  * Dot wild card selector `'.*'`.
+  * Index selector `'[<index>]'`, where `<index>` is either an (possibly negative) array index or an object member name.
+  * Index wild card selector `'[*]'`.
+  * Array slice selector `'[<start>:<end>:<step>]'`, where `<start>`, `<end>`, `<step>` are integer literals.
+  * Nested descendants selector `'..'`.
+  * Union selector `'[<sel1>,<sel2>,...,<selN>]'`, holding a comma delimited list of selectors.
+  * Filter selector `[?(<expr>)]`
+  * Current item selector `'@'`
+
+### Root Selector
 
 #### Syntax
 {: numbered="false" toc="exclude"}
 
-A dot child selector has a member name known as a dot child name or a single asterisk
-(`*`).
-
-A dot child name corresponds to a name in a object.
+Every valid JSONPath query MUST begin with the root selector named `'$'`.
 
 ~~~~ abnf
-selector = dot-child              ; see below for alternatives
-dot-child = "." dot-child-name / ; .<dot-child-name>
-            "." "*"             ; .*
-dot-child-name = 1*(
-                   "-" /         ; -
-                   DIGIT /
-                   ALPHA /
-                   "_" /         ; _
-                   %x80-10FFFF    ; any non-ASCII Unicode character
-                 )
-DIGIT =  %x30-39                  ; 0-9
-ALPHA = %x41-5A / %x61-7A         ; A-Z / a-z
-~~~~
-
-More general child names, such as the empty string, are supported by "Union
-Child" ({{unionchild}}{: format="default"}).
-
-Note that the `dot-child-name` rule follows the philosophy of JSON strings and is
-allowed to contain bit sequences that cannot encode Unicode characters (a
-single unpaired UTF-16 surrogate, for example).
-The behaviour of an implementation is undefined for child names which do
-not encode Unicode characters.
-
+root-selector  = %x24                 ; $
+~~~~
 
 #### Semantics
 {: numbered="false" toc="exclude"}
 
-A dot child name which is not a single asterisk (`*`) is considered to
-have a member name.
-It selects the value corresponding to the name from any object node.
-It selects
-no nodes from a node which is not a object.
-
-The member name of a dot child name is the sequence of Unicode characters contained
-in that name.
-
-A dot child name consisting of a single asterisk is a wild card. It selects
-all the values of any object node.
-It also selects all the elements of any array node.
-It selects no nodes from
-number, string, or literal nodes.
+The Argument &mdash; the root JSON value &ndash; is anonymous by nature. By getting assigned the universal name `'$'` it becomes the root node.
```

Comment:
Writing the actual characters instead of HTML entity references is preferred.

The same dashes should be used at the start and at the end of an interjection.